components/builder-db/src/models/package.rs

```diff
@@ -93,24 +91,20 @@ pub const ALL_COLUMNS: AllColumns = (
     origin_packages::visibility,
     origin_packages::created_at,
     origin_packages::updated_at,
+    origin_packages::origin,
 );
 
 type All = diesel::dsl::Select<origin_packages::table, AllColumns>;
-#[derive(Debug, Serialize, Deserialize, QueryableByName)]
-pub struct PackagePlatform {
-    #[sql_type = "Text"]
-    pub target: String,
-}
 
-// TED TODO - Remove this and derive AsChangeset above
-#[derive(Debug, Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Deserialize, Clone, Insertable)]
+#[table_name = "origin_packages"]
 pub struct NewPackage {
-    #[serde(with = "db_id_format")]
-    pub origin_id: i64,
+    pub origin: String,
     #[serde(with = "db_id_format")]
     pub owner_id: i64,
     pub name: String,
     pub ident: BuilderPackageIdent,
+    pub ident_array: Vec<String>,
```

Comment:
Having both ident and ident array seems redundant?

Reply:
Previously, we were splitting on slashes at insertion time. Now, if we want to insert that array we have to split it outside. Also, if we want to use insertable we have to have all columns present in the struct that we want to set. 

So yes, it is redundant but necessary. We might be able to do an audit to see where we're using ident in the code now to see if we could drop that column in the future.